File: SearchInRotatedSortedArray.java
int search(int[] arr, int key) {
        // Complete this function
        int l = 0;
        int r = arr.length - 1;
        while(l <= r){
            int m = (l + r) / 2;
            if(arr[m] == key){
                return m;
            }
            
            if(arr[m] >= arr[l]){
                if(key >= arr[l] && key <= arr[m]){
                   r = m - 1; 
                }
                else{
                    l = m + 1;
                }
            }
            else{
                if(key>= arr[m] && key <= arr[r]){
                    l = m + 1;
                }
                else{
                    r = m - 1;
                }
            }
        }
        return -1;
    }

File: SearchInRotatedSortedArray.cpp
int search(vector<int>& arr, int key) {
        // complete the function here
        int l = 0;
        int r = arr.size() - 1;
        while(l <= r){
            int mid = (l + r) / 2;
            if(arr[mid] == key){
                return mid;
            }
            if(arr[mid] >= arr[l]){
                if(key >= arr[l] && key <= arr[mid]){
                    r = mid - 1;
                }
                else{
                    l = mid + 1;
                }
            }
            else{
                if(key >= arr[mid] && key <= arr[r]){
                    l = mid + 1;
                }
                else{
                    r = mid - 1;
                }
            }
        }
        return -1;
    }
